src/tree.rs

```diff
@@ -444,6 +444,49 @@ impl<T> Tree<T> {
         Result::Ok(())
     }
 
+    pub fn replace(&mut self, node_id: &NodeId, mut new_node: Node<T>) -> Result<Node<T>, NodeIdError>
+    {
+        use std::mem;
+
+        let (is_valid, error) = self.is_valid_node_id(node_id);
+        if !is_valid {
+            return Result::Err(error.expect("Tree::replace: Missing an error value on finding an invalid NodeId."));
+        }
+
+        // lets cheat a bit and return new_node ;)
+        // TODO, while this is elegant, this worksaround NodeBuilder's Parameters
+        // Should we drain the old children array instead?
+        mem::swap(self.get_mut_unsafe(node_id).data_mut(), new_node.data_mut());
```

Comment:
Well all `mem` functions are not exactly ergonomic.
I think its okay to provide a convenience function in that case.